src/PrestaShopBundle/Form/Admin/Type/SearchProductType.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use PrestaShopBundle\Form\Admin\Sell\Product\SearchedProductItemType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Initiates input with ability to search for any type of product
+ */
+class SearchProductType extends EntitySearchInputType
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var string
+     */
+    private $languageIsoCode;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        RouterInterface $router,
+        string $employeeIsoCode
+    ) {
+        parent::__construct($translator);
+        $this->router = $router;
+        $this->languageIsoCode = $employeeIsoCode;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        parent::configureOptions($resolver);
+
+        $resolver->setDefaults([
+            'required' => false,
+            'label' => false,
+            'remote_url' => $this->router->generate('admin_products_v2_search_combinations', [
```

Comment:
As I suggested in another comment, maybe e could add an option to search for either a product or a specific combination? This form type could then be used in many use cases (third option would be to completely customise the search end point, but it can be added later if needed)

Or maybe actually to avoid complex options, just allowing to override `remote_url` would be enough? 🤔

Reply:
remote_url is set in defaults and defined in parent, so you can always override it